test/common/common/utility_test.cc

```diff
@@ -41,6 +41,40 @@ TEST(StringUtil, rtrim) {
   }
 }
 
+TEST(StringUtil, strlcpy) {
+  {
+    char dest[6];
+    size_t result;
+    result = StringUtil::strlcpy(dest, std::string{"hello"}.c_str(), sizeof(dest));
+    EXPECT_STREQ("hello", dest);
+    EXPECT_EQ(5U, result);
```

Comment:
nit: I would just put the EXPECT_EQ call directly around the strlcpy call since you don't use result anywhere else.